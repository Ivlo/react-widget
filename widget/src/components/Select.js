import React from "react";

// eslint-disable-next-line react/display-name
const Select = React.memo(({ handleChange, optionsData, optionDefault, optionsText }) => {
  return (
    <select onChange={handleChange} className="select">
      <>
        <option value="" selected="true" disabled="disabled">
          {optionDefault}
        </option>
        {optionsData.map((option) => (
          <option key={option.instalment_count} value={option.instalment_fee.string}>
            {`${option.instalment_count} ${optionsText} ${option.instalment_total.string} mes`}
          </option>
        ))}
      </>
    </select>
  );
});

export default Select;
