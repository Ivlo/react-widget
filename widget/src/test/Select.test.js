import React from "react";
import { mount } from "enzyme";
import Select from "../components/Select";

const optionsData = [
  {
    instalment_count: "3",
    instalment_fee: {
      value: "500",
    },
    instalment_total: {
      value: "5500",
    },
  },
  {
    instalment_count: "6",
    instalment_fee: {
      value: "500",
    },
    instalment_total: {
      value: "3000",
    },
  },
];
const handleChange = jest.fn();
const optionDefaultText = "option default text";
const optionsText = "cuotas de";
const firstOptionText = `3 ${optionsText} 55,00 € mes`;
const secondOptionText = `6 ${optionsText} 30,00 € mes`;

describe("Select", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = mount(
      <Select
        handleChange={handleChange}
        optionsData={optionsData}
        optionDefault={optionDefaultText}
        optionsText={optionsText}
      />,
    );
  });

  it("should render correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });
  it("Should render with two options inside", () => {
    const options = wrapper.find("option");
    expect(options.length).toEqual(2);
  });
  it("when it change the handleChange prop should be executed", () => {
    wrapper.simulate("change");
    expect(handleChange).toHaveBeenCalled();
  });
  it("the text of the first option should be equal to firstOptionText", () => {
    const firstOption = wrapper.find("option").at(0);
    expect(firstOption.text()).toEqual(firstOptionText);
  });

  it("the text of the second option should be equal to secondOptionText", () => {
    const firstOption = wrapper.find("option").at(1);
    expect(firstOption.text()).toEqual(secondOptionText);
  });
});
