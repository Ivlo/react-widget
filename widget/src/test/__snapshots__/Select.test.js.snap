// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select should render correctly 1`] = `
<Memo()
  handleChange={[MockFunction]}
  optionDefault="option default text"
  optionsData={
    Array [
      Object {
        "instalment_count": "3",
        "instalment_fee": Object {
          "value": "500",
        },
        "instalment_total": Object {
          "value": "5500",
        },
      },
      Object {
        "instalment_count": "6",
        "instalment_fee": Object {
          "value": "500",
        },
        "instalment_total": Object {
          "value": "3000",
        },
      },
    ]
  }
  optionsText="cuotas de"
>
  <select
    className="select"
    onChange={[MockFunction]}
  >
    <option
      key="3"
      value="500"
    >
      3 cuotas de 55,00 € mes
    </option>
    <option
      key="6"
      value="500"
    >
      6 cuotas de 30,00 € mes
    </option>
  </select>
</Memo()>
`;
